### PR TITLE
update doc for list partition table

### PR DESCRIPTION
**Pull request by Joyinqin**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

- revert https://github.com/pingcap/docs-cn/pull/5293
- update doc for the related session variable. (`tidb_enable_list_partition` )

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/5653
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: partitioned-table.md [hunk 1]
```diff
@@ -10,9 +10,9 @@ This document introduces TiDB's implementation of partitioning.
 
 ## Partitioning types
 
-This section introduces the types of partitioning which are available in TiDB. Currently, TiDB supports [Range partitioning](#range-partitioning) and [Hash partitioning](#hash-partitioning).
+This section introduces the types of partitioning which are available in TiDB. Currently, TiDB supports [Range partitioning](#range-partitioning), [List partitioning](#list-partitioning), [List COLUMNS partitioning](#list-columns-partitioning), and [Hash partitioning](#hash-partitioning).
```
Comment (qiancai) [inline partitioned-table.md:13]:
```suggestion
This section introduces the types of partitioning in TiDB. Currently, TiDB supports [Range partitioning](#range-partitioning), [List partitioning](#list-partitioning), [List COLUMNS partitioning](#list-columns-partitioning), and [Hash partitioning](#hash-partitioning).
```

File: partitioned-table.md [hunk 2]
```diff
@@ -164,6 +164,192 @@ Range partitioning is particularly useful when one or more of the following cond
...
+
+Before creating a List partitioned table, you need to set the value of the session variable `tidb_enable_table_partition` to `ON`.
+
+{{< copyable "sql" >}}
+
+```sql
+set @@session.tidb_enable_table_partition = ON
+```
+
+List partitioning is similar to Range partitioning. Unlike Range partitioning, in List partitioning, each partition contains rows for which the partitioning expression value belongs to a given value list. This value list defined for each partition has any number of values but cannot have duplicate values. You can use the `PARTITION ... VALUES IN (...)` clause to define these value lists.
```
Comment (qiancai) [inline partitioned-table.md:181]:
```suggestion
List partitioning is similar to Range partitioning. Unlike Range partitioning, in List partitioning, the partitioning expression values for all rows in each partition are in a given value set. This value set defined for each partition has any number of values but cannot have duplicate values. You can use the `PARTITION ... VALUES IN (...)` clause to define these value sets.
```

File: partitioned-table.md [hunk 3]
```diff
@@ -164,6 +164,192 @@ Range partitioning is particularly useful when one or more of the following cond
...
+
+You can also execute `ALTER TABLE employees DROP PARTITION pEast` to delete all related rows, but this statement also deletes the `pEast` partition from the table definition. In this situation, you must execute the `ALTER TABLE ... ADD PARTITION` statement to recover the original partitioning scheme of the table.
+
+Unlike Range partitioning, List partitioning does not have a similar `MAXVALUE` partition to store all values that do not belong to other partitions. Instead, all expected values of the partition expression must be included in the `PARTITION ... VALUES IN (...)` clause. If the value to be inserted in an `INSERT` statement does not match the value of any partition, the statement fails to execute and an error is reported. See the following example:
```
Comment (qiancai) [inline partitioned-table.md:228]:
```suggestion
Unlike Range partitioning, List partitioning does not have a similar `MAXVALUE` partition to store all values that do not belong to other partitions. Instead, all expected values of the partition expression must be included in the `PARTITION ... VALUES IN (...)` clause. If the value to be inserted in an `INSERT` statement does not match the column value of any partition, the statement fails to execute and an error is reported. See the following example:
```

Comment (qiancai) [inline partitioned-table.md:228]:
```suggestion
Unlike Range partitioning, List partitioning does not have a similar `MAXVALUE` partition to store all values that do not belong to other partitions. Instead, all expected values of the partition expression must be included in the `PARTITION ... VALUES IN (...)` clause. If the value to be inserted in an `INSERT` statement does not match the column value of any partition, the statement fails to execute and an error is reported. See the following example:
```

**Review by crazycs520 — COMMENTED** [review]
LGTM

File: partitioned-table.md [hunk 4]
```diff
@@ -164,6 +164,192 @@ Range partitioning is particularly useful when one or more of the following cond
...
+    ->   PARTITION p0 VALUES IN (1, 2, 3),
+    ->   PARTITION p1 VALUES IN (4, 5, 6)
+    -> );
+Query OK, 0 rows affected (0.11 sec)
+
+test> INSERT INTO t VALUES (7, 7);
+ERROR 1525 (HY000): Table has no partition for value 7
+```
+
+To ignore the error type above, you can use the `IGNORE` keyword. After using this keyword, if a row contains values that do not match the partition's value lists, this row will not be inserted. Instead, any row with matched values is inserted, and no error is reported:
```
Comment (qiancai) [inline partitioned-table.md:245]:
```suggestion
To ignore the error type above, you can use the `IGNORE` keyword. After using this keyword, if a row contains values that do not match the column value set of any partition, this row will not be inserted. Instead, any row with matched values is inserted, and no error is reported:
```

File: partitioned-table.md [hunk 5]
```diff
@@ -164,6 +164,192 @@ Range partitioning is particularly useful when one or more of the following cond
...
+
+You can also execute `ALTER TABLE employees DROP PARTITION pEast` to delete all related rows, but this statement also deletes the `pEast` partition from the table definition. In this situation, you must execute the `ALTER TABLE ... ADD PARTITION` statement to recover the original partitioning scheme of the table.
+
+Unlike Range partitioning, List partitioning does not have a similar `MAXVALUE` partition to store all values that do not belong to other partitions. Instead, all expected values of the partition expression must be included in the `PARTITION ... VALUES IN (...)` clause. If the value to be inserted in an `INSERT` statement does not match the column value of any partition, the statement fails to execute and an error is reported. See the following example:
```
Comment (qiancai) [inline partitioned-table.md:228]:
```suggestion
Unlike Range partitioning, List partitioning does not have a similar `MAXVALUE` partition to store all values that do not belong to other partitions. Instead, all expected values of the partition expression must be included in the `PARTITION ... VALUES IN (...)` clause. If the value to be inserted in an `INSERT` statement does not match the column value set of any partition, the statement fails to execute and an error is reported. See the following example:
```

File: partitioned-table.md [hunk 6]
```diff
@@ -949,7 +1135,7 @@ YEARWEEK()
 
 ### Compatibility with MySQL
 
-Currently, TiDB only supports Range partitioning and Hash partitioning. Other partitioning types that are available in MySQL such as key partitioning are not supported yet in TiDB.
+Currently, TiDB only supports Range partitioning, List partitioning, List COLUMNS partitioning and Hash partitioning. Other partitioning types that are available in MySQL such as key partitioning are not supported yet in TiDB.
```
Comment (qiancai) [inline partitioned-table.md:1138]:
```suggestion
Currently, TiDB supports Range partitioning, List partitioning, List COLUMNS partitioning, and Hash partitioning. Other partitioning types that are available in MySQL such as key partitioning are not supported yet in TiDB.
```

File: partitioned-table.md [hunk 7]
```diff
@@ -1049,6 +1235,6 @@ select * from t;
 5 rows in set (0.00 sec)
 ```
 
-The `tidb_enable_table_partition` environment variable controls whether to enable the partitioned table feature. If this variable is set to `off`, the partition information will be ignored when a table is created, and this table will be created as a normal table.
+The `tidb_enable_table_partition` environment variable controls whether to enable the partitioned table feature. If this variable is set to `OFF`, the partition information will be ignored when a table is created, and this table will be created as a normal table.
```
Comment (qiancai) [inline partitioned-table.md:1238]: According to the original PR, the variable name here is changed to `tidb_enable_list_partition`.
```suggestion
The `tidb_enable_list_partition` environment variable controls whether to enable the partitioned table feature. If this variable is set to `OFF`, the partition information will be ignored when a table is created, and this table will be created as a normal table.
```

File: system-variables.md [hunk 8]
```diff
@@ -462,11 +462,20 @@ Constraint checking is always performed in place for pessimistic transactions (d
 - Scope: SESSION | GLOBAL
 - Default value: ON
 - This variable is used to set whether to enable the `TABLE PARTITION` feature:
-
     - `ON` indicates enabling Range partitioning, Hash partitioning, and Range column partitioning with one single column.
     - `AUTO` functions the same way as `ON` does.
     - `OFF` indicates disabling the `TABLE PARTITION` feature. In this case, the syntax that creates a partition table can be executed, but the table created is not a partitioned one.
 
+### `tidb_enable_list_partition` <span class="version-mark">New in v5.0 GA</span>
```
Comment (qiancai) [inline system-variables.md:469]:
```suggestion
### tidb_enable_list_partition <span class="version-mark">New in v5.0 GA</span>
```

File: partitioned-table.md [hunk 9]
```diff
@@ -164,6 +164,192 @@ Range partitioning is particularly useful when one or more of the following cond
...
+3 rows in set (0.01 sec)
+```
+
+### List COLUMNS partitioning
+
+> **Warning:**
+>
+> List COLUMNS partitioning is an experimental feature. It is not recommended that you use it in the production environment.
+
+List COLUMNS partitioning is a variant of List partitioning. You can use multiple columns as partition keys and use columns of the data types other than integer as partition columns. You can also use columns of the string, `DATE`, and `DATETIME` types.
```
Comment (qiancai) [inline partitioned-table.md:272]: According to the confirmation from @crazycs520, List COLUMNS also supports the integer type, so "other than" in this paragraph is confusing. 

```suggestion
List COLUMNS partitioning is a variant of List partitioning. You can use multiple columns as partition keys. Besides the integer data type, you can also use the columns in the string, `DATE`, and `DATETIME` data types as partition columns.
```

**Comment by qiancai** [comment]
/ label requires-version-specific-changes

File: system-variables.md [hunk 10]
```diff
@@ -462,11 +462,20 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
-
     - `ON` indicates enabling Range partitioning, Hash partitioning, and Range column partitioning with one single column.
     - `AUTO` functions the same way as `ON` does.
     - `OFF` indicates disabling the `TABLE PARTITION` feature. In this case, the syntax that creates a partition table can be executed, but the table created is not a partitioned one.
 
+### tidb_enable_list_partition <span class="version-mark">New in v5.0 GA</span>
+
+> **Warning:**
+>
+> Currently, List partition and List COLUMNS partition are experimental features and you are not recommended to use them in a production environment.
```
Comment (TomShawn) [inline system-variables.md:473]:
```suggestion
> Currently, List partition and List COLUMNS partition are experimental features. It is not recommended to use them in a production environment.
```

**Pull request merged**